innovation_sweet_spots/tests/data_update_checks/utils.py

```diff
@@ -0,0 +1,193 @@
+
+
+def find_filenames(
+    path: pathlib.Path,
+) -> list:
+    """Return a set of filenames from path"""
+    return {file.name for file in path.iterdir() if file.name != ".DS_Store"}
+
+
+def check_filenames_match(cur_files: set, new_files: set, cur_dir: str, new_dir: str):
+    """Check that current and new sets of filenames are the same.
+    Log the outcome and provide additional information on the
+    differences."""
+    logger.info("Checking filenames match...")
+    if cur_files == new_files:
+        logger.info(f"{cur_dir}/ and {new_dir}/ files match.")
+    else:
+        logger.warning(f"{cur_dir}/ and {new_dir}/ files do not match.")
+        if missing_files := cur_files - new_files:
+            logger.warning(
+                f"Compared with {cur_dir}/, the {new_dir}/ dataset is missing {len(missing_files)} files: {missing_files}."
+            )
+        if add_files := new_files - cur_files:
+            logger.warning(
+                f"Compared with {cur_dir}/, the {new_dir}/ dataset has {len(add_files)} additional files: {add_files}."
+            )
+    print("\n")
+
+
+def check_n_rows(
+    cur_df: pd.DataFrame,
+    new_df: pd.DataFrame,
+    filename: str,
+    cur_dir: str,
+    new_dir: str,
+):
+    """Check that the number of rows in the current and new dataframes are
+    the same. Log the outcome and provide additional information on the
+    differences."""
+    additional_rows = len(new_df) - len(cur_df)
+    start_message = f"The {new_dir}/ version of the file {filename} has"
+    end_message = f"rows of data compared to the {cur_dir}/ version"
+    if additional_rows > 0:
+        logger.info(f"{start_message} {additional_rows} additional {end_message}.")
+    elif additional_rows < 0:
+        logger.warning(f"{start_message} {additional_rows} fewer {end_message}.")
+    else:
+        logger.warning(f"{start_message} the same number of {end_message}.")
+
+
+def check_cols_match(
+    cur_df: pd.DataFrame,
+    new_df: pd.DataFrame,
+    filename: str,
+    cur_dir: str,
+    new_dir: str,
+):
+    """Check that the number of columns in the current and new dataframes are
+    the ame. Log the outcome and provide additional information on the
+    differences."""
+    cur_cols = set(cur_df.columns)
+    new_cols = set(new_df.columns)
+    if cur_cols == new_cols:
+        logger.info(f"{filename} in {cur_dir}/ and {new_dir}/ columns match.")
+    else:
+        logger.warning(f"{filename} in {cur_dir}/ and {new_dir}/ columns do not match.")
+        if missing_cols := cur_cols - new_cols:
+            logger.warning(
+                f"Compared with {cur_dir}/{filename}, {new_dir}/{filename} dataset is missing {len(missing_cols)} columns: {missing_cols}."
+            )
+        if add_cols := new_cols - cur_cols:
+            logger.warning(
+                f"Compared with {cur_dir}/{filename}, {new_dir}/{filename} dataset has {len(add_cols)} additional columns: {add_cols}."
+            )
```

Comment:
I wonder if it would be useful here to also list the names of the extra or missing columns?